src/commands.c

```diff
@@ -2240,9 +2240,15 @@ void cmd_gaps(I3_CMD, const char *type, const char *scope, const char *mode, con
     }
 
     if (!strcmp(type, "inner")) {
-        CMD_GAPS(inner, outer);
-    } else if (!strcmp(type, "outer")) {
-        CMD_GAPS(outer, inner);
+        CMD_GAPS(inner);
+    } else if (!strcmp(type, "top") || !strcmp(type, "vertical") || !strcmp(type, "outer")) {
```

Comment:
Actually I'm just stumbling across this… this logic here isn't correct anymore, is it? When using "vertical", only the "top" one will be set. Since it's else-ifs, the bottom one will never be set. This mistake is on me due to my review comment, sorry. I guess we need to go back to the boolean flag here. Oops :-)

Reply:
Ahh, good catch. I only briefly looked at that code and took your suggestion as is. I made sure it built, but I didn't restart my entire X Session to test it :no_mouth: Let me do that now.

Reply:
I just pushed that change. Since the additional `CMD_GAPS` are now in the same scope, I pulled out the declaration of `current_value` and `reset`.

Reply:
I think it'd be better if we instead used the generally recommend pattern for macros:

```
#define …      \
  do {         \
    // …       \
  } while (0); \
```

That way we also don't have any scoping issues.

Reply:
OK, I put them back and made that change.

Reply:
Thanks again for the patience. I think we are *finally* done now. Waiting for Travis to come back OK, then I'll merge it.